src/foam/nanos/http/AuthWebAgent.java

```diff
@@ -168,8 +170,19 @@ public Session authenticate(X x) {
           .put(HttpServletResponse.class, resp), email, password);
 
         if ( user != null ) {
+          // if user is attempting to and (Can)actAs another entity, set the entity in session context
+          if ( ! SafetyUtil.isEmpty(entityId) ) {
+            AgentAuthService agentService = (AgentAuthService) x.get("agentAuth");
+            DAO localUserDAO = (DAO) x.get("localUserDAO");
+            User entity = (User) localUserDAO.find(Long.parseLong(entityId));
+            if (agentService.canActAs(x, user, entity)) {
+              // set agent in session
+              session.setContext(session.getContext().put("entity",entity));
+            }
+          }
           return session;
-        } else {
+          }
```

Comment:
This was correct before.

Reply:
Sorry, not sure what you mean by this @TW80000 . Fixed the rest of the styling issues.